synfig-studio/src/gui/app.cpp

```diff
@@ -1364,7 +1364,11 @@ DEFINE_ACTION("keyframe-properties", _("Properties"));
 	ACCEL("<Control>9",								"<Actions>/canvasview/quality-09"					);
 	ACCEL("<Control>0",								"<Actions>/canvasview/quality-10"					);
 	ACCEL("<Control>z",								"<Actions>/action_group_dock_history/undo"				);
-	ACCEL("<Control>r",								"<Actions>/action_group_dock_history/redo"				);
+#if __gnu_linux__
+	ACCEL("<Control><Shift>Z",                      "<Actions>/action_group_dock_history/redo");
```

Comment:
Is any difference here between lower-case 'z' and upper-case 'Z'? Asking because as I can see lower-case is used before.

Reply:
Uh no...

I used upper case because of my habit with newer GTKMM versions. Lower case doesn't work but in this PR both works OK. I'm compiling against newer GTKMM 3.24 so...

But if you want consistency I could change that :)

Reply:
I think that's okay then, because we'll come back to this code again. (For Cmd/Ctrl key for MacOS)

Reply:
Wouldn't be better to use `#if defined(_WIN32) || defined(__APPLE__)`?
I don't know if `__gnu_linux__` is for clang compiler as well.
A random article says `__linux__` would be enough, but I don't know :P
https://blog.kowalczyk.info/article/j/guide-to-predefined-macros-in-c-compilers-gcc-clang-msvc-etc..html

Reply:
@rodolforg You are right. For MacOS redo is Cmd+Shift+Z (https://osxdaily.com/how-undo-redo-mac/), so using `#ifdef _WIN32` is a way  better.

Reply:
@Keyikedalube maybe also check `<Primary>` key here? (for Cmd/Ctrl+Shit+Z) If you make a change, I can test it on MacOS.

Reply:
@rodolforg I referenced header macro from this website https://sourceforge.net/p/predef/wiki/OperatingSystems/

Reply:
@ice0 <Primary> works on my side. It's mapped to Control key for Linux

Reply:
@Keyikedalube your link also says `__linux__` is valid XD.
Anyway, I believe it's better to use synfig 'standard' way: first the windows case, else (if necessary) apple case, else anything else (linux, openbsd, etc.)

@ice0 why `<Primary>` here only? Mac user will be confused: `ctrl+z` undo, `ctrl+y` or `ctrl+shift+z` does not anything.

Reply:
@rodolforg just for testing purposes. Just checked - all works as expected (on MacOS <Primary> will become Command key). So I suggest 2 changes to this PR.
1) Change ifdef as suggested by Rodolfo:
```
#if _WIN32
	ACCEL("<Control>y",                             "<Actions>/action_group_dock_history/redo");
#else
	ACCEL("<Primary><Shift>z",                      "<Actions>/action_group_dock_history/redo");
#endif
```
2. Replace <Control> to <Primary> for Undo (Ctrl+Z)

After that we can switch all other shortcuts.
File:
- New (Ctrl+N)
- Open (Ctrl+O)
- Save (Ctrl+S)
- Save As (Shift+Ctrl+S)
- Import (Ctrl+I)
- New (Ctrl+N)

Edit:
- Select All Layers (Shift+Ctrl+A)
- Unselect All Layers (Shift+Ctrl+D)
- Select All Handles (Ctrl+A)
- Unselect All Handles (Ctrl+D)

etc...

Reply:
Working on it...